node/src/command.rs

```diff
@@ -268,25 +263,28 @@ pub fn run() -> Result<()> {
 			let runner = cli.create_runner(cmd)?;
 			// Switch on the concrete benchmark sub-command-
 			match cmd {
-				BenchmarkCmd::Pallet(cmd) =>
+				BenchmarkCmd::Pallet(cmd) => {
 					if cfg!(feature = "runtime-benchmarks") {
-						runner.sync_run(|config| cmd.run::<Block, ()>(config))
+						runner.sync_run(|config| {
+							cmd.run_with_spec::<HashingFor<Block>, ReclaimHostFunctions>(Some(
+								config.chain_spec,
+							))
+						})
 					} else {
 						Err("Benchmarking wasn't enabled when building the node. \
 					You can enable it with `--features runtime-benchmarks`."
 							.into())
-					},
+					}
+				},
 				BenchmarkCmd::Block(cmd) => runner.sync_run(|config| {
 					construct_benchmark_partials!(config, |partials| cmd.run(partials.client))
 				}),
```

Comment:
In the `base-parachain` and SDK:
```suggestion
				BenchmarkCmd::Block(cmd) => runner.sync_run(|config| {
					let partials = new_partial(&config)?;
					cmd.run(partials.client)
				}),
```

Reply:
@Daanvdplas  This code block is actually the same, but pop-node uses a special macro to dynamically support testnet and devnet. 
```
macro_rules! construct_benchmark_partials {
	($config:expr, |$partials:ident| $code:expr) => {
		match $config.chain_spec.runtime() {
			Runtime::Devnet => {
				let $partials = new_partial::<pop_runtime_devnet::RuntimeApi>(&$config)?;
				$code
			},
			Runtime::Testnet => {
				let $partials = new_partial::<pop_runtime_testnet::RuntimeApi>(&$config)?;
				$code
			},
		}
	};
}
```

Reply:
Thank you!